unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/UIHelpers/HUDCameraCanvasHelper.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+/// <summary>
+/// This helper will prepare the canvas to work in screen space or camera space if the proper DataStore is set
+/// </summary>
+public class HUDCameraCanvasHelper : IDisposable
+{
+    internal readonly Canvas canvas;
+    internal readonly BaseVariable<Camera> hudCameraVariable;
+
+    public HUDCameraCanvasHelper(Canvas canvas, BaseVariable<Camera> hudCameraVariable)
```

Comment:
would it be better to be a component (prefab configuration) instead of instancing it manually on the code every ui?

Reply:
You need the UI camera, which might not be ready by the time you awake/start your monobehaviour. In the end you would have to do the same stuff you are doing here in every HUD, subscribing to changes in the store and changing your canvas settings.

I find more convenient a Composition approach encapsulating that behaviour here.